### DISCUSSION TITLE
terraform flux_bootstrap_git failed

### DISCUSSION THREAD

**Discussion by pengliaoye** [fluxcd/flux2#5030] [Q&A]
Error: Bootstrap run error
│
│   with flux_bootstrap_git.this,
│   on main.tf line 291, in resource "flux_bootstrap_git" "this":
│  291: resource "flux_bootstrap_git" "this" {
│
│ timeout waiting for: [CustomResourceDefinition/helmreleases.helm.toolkit.fluxcd.io status: 'Terminating']

**Comment by stefankyriacou123** [comment]
In case its useful to anyone, I had this same issue and it turned out that terraform had uninstalled the flux controller before uninstalling a failed HelmRelease crd which still had a finalizer that couldn't be run, meaning k8s couldn't mark "helmrelease.helm.toolkit.fluxcd.io" as being uninstalled, so it was perpetually stuck in a "Terminating" state.

I ended up fixing the issue by identifying the HelmRelease that had failed to install with the following command:
`kubectl get helmreleases -A`

Then patching the listed HelmRelease/s to remove the finalizers (if you have more than one, then I assume you will have to patch all of them) with the following command:
`kubectl patch hr <HELMRELEASE NAME> -n <NAMESPACE> -p '{"metadata":{"finalizers":[]}}' --type=merge`

Once I did that, k8s automatically removed all the last lingering stuff from the previous install of flux, and running `terraform apply` again worked successfully.
